### ISSUE TITLE
IT in `prod` fails due to ConnectionResetError

### ISSUE THREAD

**Issue by achave11** [DataBiosphere/azul#4186]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/20802
```
======================================================================
ERROR: test_indexing (integration_test.IndexingIntegrationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 445, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 440, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.8/http/client.py", line 1348, in getresponse
    response.begin()
  File "/usr/local/lib/python3.8/http/client.py", line 316, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.8/http/client.py", line 277, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.8/ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "/usr/local/lib/python3.8/ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
ConnectionResetError: [Errno 104] Connection reset by peer
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/azul_test_case.py", line 172, in wrapped
    return method(*args, **kwargs)
  File "/builds/ucsc/azul/test/integration_test.py", line 420, in test_indexing
    notifications, fqids = self._prepare_notifications(catalog)
  File "/builds/ucsc/azul/test/integration_test.py", line 862, in _prepare_notifications
    list(starmap(update, self._list_bundles(catalog,
  File "/builds/ucsc/azul/test/integration_test.py", line 317, in _list_bundles
    source, prefix, new_fqids = self._list_partition_bundles(catalog, source)
  File "/builds/ucsc/azul/test/integration_test.py", line 254, in _list_partition_bundles
    source = plugin.resolve_source(source)
  File "/builds/ucsc/azul/src/azul/plugins/__init__.py", line 368, in resolve_source
    id = self.lookup_source_id(spec)
  File "/builds/ucsc/azul/src/azul/plugins/repository/tdr/__init__.py", line 246, in lookup_source_id
    return self.tdr.lookup_source(spec).id
  File "/builds/ucsc/azul/src/azul/terra.py", line 386, in lookup_source
    source = self._lookup_source(source_spec)
  File "/builds/ucsc/azul/src/azul/terra.py", line 409, in _lookup_source
    response = self._request('GET', endpoint, fields=params)
  File "/builds/ucsc/azul/src/azul/terra.py", line 300, in _request
    response = self._http_client.request(method,
  File "/build/.venv/lib/python3.8/site-packages/urllib3/request.py", line 74, in request
    return self.request_encode_url(
  File "/build/.venv/lib/python3.8/site-packages/urllib3/request.py", line 96, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/build/.venv/lib/python3.8/site-packages/google/auth/transport/urllib3.py", line 388, in urlopen
    response = self.http.urlopen(
  File "/build/.venv/lib/python3.8/site-packages/urllib3/poolmanager.py", line 375, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/build/.venv/lib/python3.8/site-packages/urllib3/util/retry.py", line 507, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/build/.venv/lib/python3.8/site-packages/urllib3/packages/six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 445, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/build/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 440, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.8/http/client.py", line 1348, in getresponse
    response.begin()
  File "/usr/local/lib/python3.8/http/client.py", line 316, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.8/http/client.py", line 277, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.8/ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "/usr/local/lib/python3.8/ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
----------------------------------------------------------------------
Ran 16 tests in 762.405s
FAILED (errors=1, skipped=4)
make: *** [Makefile:160: integration_test] Error 1
Cleaning up project directory and file based variables 00:01
ERROR: Job failed: exit code 1
```

**Comment by achave11** [comment]
The last log output from `test_indexing` …
```
2022-05-12 17:58:43,075 DEBUG   MainThread: _request(…) -> b'{"total":329,"filteredTotal":329,"items":[{"id":"6d39533c-1f7f-494e-9cee-40ed2f801e19","name":"hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dcp2_20220121_dcp12","description":"Create snapshot hca_prod_d012d4768f8c4ff389d6ebbe22c1b5c1__20220119_dc...'
2022-05-12 17:58:43,077 DEBUG   MainThread: _request('GET', furl('https://data.terra.bio/api/repository/v1/snapshots'), fields={'offset': 329, 'limit': 200, 'sort': 'created_date', 'direction': 'asc'}, headers=None, timeout=20.0, body=None)
2022-05-12 17:58:43,210 DEBUG   MainThread: _request(…) -> b'{"total":329,"filteredTotal":329,"items":[],"roleMap":{}}'
```

**Comment by achave11** [comment]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/21146

**Comment by achave11** [comment]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/21460

**Comment by achave11** [comment]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/21466

**Comment by achave11** [comment]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/21467

**Comment by achave11** [comment]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/21496

**Comment by achave11** [comment]
https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/21504

**Comment by melainalegaspi** [comment]
See https://humancellatlas.slack.com/archives/C01360XN04S/p1653531046662249

**Comment by melainalegaspi** [comment]
@hannes-ucsc : "I fixed this in a593e166. I just need to make sure if I backported that hotfix."

**Comment by hannes-ucsc** [comment]
Yes, backport is 5f1b30da.
